packages/babel-parser/src/parser/expression.js

```diff
@@ -222,40 +222,40 @@ export default class ExpressionParser extends LValParser {
   parseMaybeAssignDisallowIn(
     refExpressionErrors?: ?ExpressionErrors,
     afterLeftParse?: Function,
-    refNeedsArrowPos?: ?Pos,
   ) {
     return this.disallowInAnd(() =>
-      this.parseMaybeAssign(
-        refExpressionErrors,
-        afterLeftParse,
-        refNeedsArrowPos,
-      ),
+      this.parseMaybeAssign(refExpressionErrors, afterLeftParse),
     );
   }
 
   // Set [+In] parameter for assignment expression
   parseMaybeAssignAllowIn(
     refExpressionErrors?: ?ExpressionErrors,
     afterLeftParse?: Function,
-    refNeedsArrowPos?: ?Pos,
   ) {
     return this.allowInAnd(() =>
-      this.parseMaybeAssign(
-        refExpressionErrors,
-        afterLeftParse,
-        refNeedsArrowPos,
-      ),
+      this.parseMaybeAssign(refExpressionErrors, afterLeftParse),
     );
   }
 
   // Parse an assignment expression. This includes applications of
   // operators like `+=`.
 
+  setOptionalParametersError(
+    refExpressionErrors: ExpressionErrors,
+    resultError?: ?EnrichedSyntaxError,
+  ) {
+    if (resultError && resultError.pos) {
+      refExpressionErrors.optionalParameters = resultError.pos;
+    } else {
+      refExpressionErrors.optionalParameters = this.state.start;
+    }
```

Comment:
Nit: this can be simplified to
```suggestion
    refExpressionErrors.optionalParameters = resultError?.pos || this.state.start;
```